WordPress/src/main/java/org/wordpress/android/ui/reader/repository/usecases/FetchInterestTagsUseCase.kt

```diff
@@ -0,0 +1,54 @@
+package org.wordpress.android.ui.reader.repository.usecases
+
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.ui.reader.ReaderEvents.InterestTagsFetchEnded
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryCommunication
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryCommunication.Error.RemoteRequestFailure
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryCommunication.SuccessWithData
+import org.wordpress.android.ui.reader.services.update.ReaderUpdateLogic.UpdateTask.INTEREST_TAGS
+import org.wordpress.android.ui.reader.services.update.ReaderUpdateServiceStarter
+import org.wordpress.android.util.EventBusWrapper
+import org.wordpress.android.viewmodel.ContextProvider
+import java.util.EnumSet
+import javax.inject.Inject
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class FetchInterestTagsUseCase @Inject constructor(
+    private val contextProvider: ContextProvider,
+    private val eventBusWrapper: EventBusWrapper
+) {
+    private var continuation: Continuation<ReaderRepositoryCommunication>? = null
+
+    suspend fun fetch(): ReaderRepositoryCommunication {
+        if (continuation != null) {
+            throw IllegalStateException("Fetch already in progress.")
+        }
+        return suspendCoroutine { cont ->
+            continuation = cont
+            eventBusWrapper.register(this)
+
+            ReaderUpdateServiceStarter.startService(
+                contextProvider.getContext(),
+                EnumSet.of(INTEREST_TAGS)
+            )
```

Comment:
I'm registering to event bus inside the use case, I've not seen this practice elsewhere in the code. Let me know if you see any downside.

Reply:
@ashiagr - There is similar logic in #12454, but I used init & stop to set up the bus to support multiple requests at the same time. Not sure if you'll need to do that here, but wanted to mention it. Also, in the same PR, I have a `ReaderRepositoryDispatchingUseCase` that I use for all the repo usecases to set up coroutine context.

Reply:
@malinajirka what do you think? Should we use `init` & `stop` pattern in this case? I'll make changes accordingly.

Reply:
Since this useCase doesn't support parallel request by its nature, I think the current implementation looks good. I don't think we need to change anything.